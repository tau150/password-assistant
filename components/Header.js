import * as React from 'react'
import Image from "next/image"
import { useRouter } from 'next/dist/client/router'
import { FormattedMessage } from 'react-intl'
import { Flex, FormControl, FormLabel, Select, Box } from '@chakra-ui/react'

function Header(){
  const router = useRouter()
  const { locale } = router

  const handleLangChange = (e) => {
    const { value: locale } = e.target
    router.push(router.asPath, router.asPath, { locale })
  }

  return (
    <Flex as='header' bgColor='blue.100' justifyContent='space-between' alignItems='center' py={4} px={8}>
      <Box width={['40%', '10%' ]}h='40px' position='relative' zIndex='2'>
        <Image layout="fill" objectFit="contain" src="https://www.openbank.es/assets/logo_topbar/Logo_Cancer_Mama_Web_3.svg" alt='openbank logo'/>
      </Box>
      <FormControl m={0} id="lang" d='flex' alignItems='center' w={['50%', '15%']}>
        <FormLabel my={0} color='brand.secondary' minW='80px'>
          <FormattedMessage id='language' defaultMessage='Language' />
        </FormLabel>
        <Select borderColor='brand.secondary' value={locale} onChange={handleLangChange} aria-label='language-selector'>
          <option value="es">ES</option>
          <option value="en">EN</option>
        </Select>
      </FormControl>
    </Flex>
  )
}
export default Header