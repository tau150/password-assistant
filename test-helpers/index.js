
import * as React from 'react'
import {   render as rtlRender } from '@testing-library/react'
import { IntlProvider } from 'react-intl';
import { ChakraProvider } from "@chakra-ui/react"

import theme from '../styles/theme';
import intlENmessages from '../content/locale/en.json'

function AppProviders({children}){
  return (
    <IntlProvider
      locale='en'
      messages={intlENmessages}
      onError={() => null}>
      <ChakraProvider theme={theme}>
        {children}
      </ChakraProvider>
    </IntlProvider>
  )
}


export function render(ui, ...renderOptions){
  const returnValue = {
    ...rtlRender(ui, {wrapper: AppProviders, ...renderOptions}),
  }
  return returnValue
}
