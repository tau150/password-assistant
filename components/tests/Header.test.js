import * as React from 'react'
import { screen } from '@testing-library/dom'
import { render } from '../../test-helpers'
import userEvent from '@testing-library/user-event'
import * as nextRouter from 'next/router'

import Header from '../Header'

const mockedPush = jest.fn()
beforeEach(() => {
  nextRouter.useRouter = jest.fn()
  nextRouter.useRouter.mockImplementation(() => ({ locale: 'en', push: mockedPush }))
})

describe('Header test', () => {
  it('renders Information Intro', () => {
    expect(render(<Header />)).toMatchSnapshot()
  })

  it('should change language', () => {
    render(<Header />)
    const langSelector = screen.getByRole('combobox', {name: 'language-selector'})
    userEvent.selectOptions(langSelector, ['es'])
    expect(mockedPush).toHaveBeenCalled()
  })
})

