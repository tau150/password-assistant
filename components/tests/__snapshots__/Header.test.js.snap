// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header test renders Information Intro 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    class="chakra-ui-light"
  >
    <div
      id="chakra-toast-portal"
    >
      <ul
        id="chakra-toast-manager-top"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
      />
      <ul
        id="chakra-toast-manager-top-left"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-top-right"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-bottom-left"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-bottom"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
      />
      <ul
        id="chakra-toast-manager-bottom-right"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
    </div>
    <div>
      <header
        class="css-1mkudia"
      >
        <div
          class="css-195ke11"
        >
          <img
            alt="test"
          />
        </div>
        <div
          class="chakra-form-control css-fweyon"
          role="group"
        >
          <label
            class="chakra-form__label css-15p4kd2"
            for="lang"
            id="lang-label"
          >
            Language
          </label>
          <div
            class="chakra-select__wrapper css-42b2qy"
          >
            <select
              aria-label="language-selector"
              class="chakra-select css-avb2le"
              id="lang"
            >
              <option
                value="es"
              >
                ES
              </option>
              <option
                value="en"
              >
                EN
              </option>
            </select>
            <div
              class="chakra-select__icon-wrapper css-zj2dqk"
            >
              <svg
                aria-hidden="true"
                class="chakra-select__icon"
                focusable="false"
                role="presentation"
                style="width: 1em; height: 1em; color: currentColor;"
                viewBox="0 0 24 24"
              >
                <path
                  d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                  fill="currentColor"
                />
              </svg>
            </div>
          </div>
        </div>
      </header>
    </div>
  </body>,
  "container": <div>
    <header
      class="css-1mkudia"
    >
      <div
        class="css-195ke11"
      >
        <img
          alt="test"
        />
      </div>
      <div
        class="chakra-form-control css-fweyon"
        role="group"
      >
        <label
          class="chakra-form__label css-15p4kd2"
          for="lang"
          id="lang-label"
        >
          Language
        </label>
        <div
          class="chakra-select__wrapper css-42b2qy"
        >
          <select
            aria-label="language-selector"
            class="chakra-select css-avb2le"
            id="lang"
          >
            <option
              value="es"
            >
              ES
            </option>
            <option
              value="en"
            >
              EN
            </option>
          </select>
          <div
            class="chakra-select__icon-wrapper css-zj2dqk"
          >
            <svg
              aria-hidden="true"
              class="chakra-select__icon"
              focusable="false"
              role="presentation"
              style="width: 1em; height: 1em; color: currentColor;"
              viewBox="0 0 24 24"
            >
              <path
                d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                fill="currentColor"
              />
            </svg>
          </div>
        </div>
      </div>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
